### PR TITLE
Fix/347: Network configs

### LINKED ISSUE #347: Support network configuration
## The Chore

What we need to achieve and who for

## Tasks

- [ ] What do we need to do first
- [ ] and then what?
- [ ] Etc.

### PR DESCRIPTION
**Pull request by notbot00**
Closes #347 

Networks live in two separate repos now, which needed to be addressed:
- https://github.com/vegaprotocol/networks
- https://github.com/vegaprotocol/networks-internal

Added separate test networks section to handle fairground (and potentially other test networks exposed in the future). Added the ability to remove a network based on the [specs](https://github.com/vegaprotocol/specs/blob/master/user-interface/0001-WALL-wallet.md#configure-network).

networks
Screenshot 2022-10-10 at 09 33 53
Screenshot 2022-10-10 at 09 35 06
Screenshot 2022-10-10 at 09 35 24

### CODE REVIEW

File: frontend/src/components/chrome/drawer-manage-network.tsx [hunk 1]
```diff
@@ -16,45 +11,17 @@ export function DrawerManageNetwork({
...
-        Add a network
-      </Header>
-      <NetworkImportForm />
-    </>
+    <div>
+      <NetworkPresets
+        setEditView={() => setView('edit')}
+        setSelectedNetwork={setSelectedNetwork}
+      />
+      <div style={{ margin: '24px 0' }}>
```
Comment (elmargasimov) [inline frontend/src/components/chrome/drawer-manage-network.tsx:20]: suggestion(non-blocking): maybe use tailwind css? You can use custom/arbitrary values as well my-[24px]

File: frontend/src/components/network-import-form/network-import-form.tsx [hunk 2]
```diff
@@ -26,11 +22,7 @@ interface NetworkImportFormProps {
 }
 
 export function NetworkImportForm({ onComplete }: NetworkImportFormProps) {
-  const [showOverwriteCheckbox, setShowOverwriteCheckbox] =
-    React.useState(false)
-  const {
-    state: { networks, presets }
-  } = useGlobal()
+  const [showOverwriteCheckbox, setShowOverwriteCheckbox] = useState(false)
```
Comment (elmargasimov) [inline frontend/src/components/network-import-form/network-import-form.tsx:25]: suggestion(non-blocking): nothing major, but conventionally you would use "is" for  binary state so e.g isCheckboxVisible

File: frontend/src/components/network-import-form/network-import-form.tsx [hunk 3]
```diff
@@ -86,81 +75,49 @@ export function NetworkImportForm({ onComplete }: NetworkImportFormProps) {
...
-    <form onSubmit={handleSubmit(submit)}>
+    <form
+      onSubmit={handleSubmit(values => submit({ ...values, network: 'other' }))}
+    >
       <FormGroup
-        label='Network'
-        labelFor='network'
-        intent={errors.network?.message ? Intent.DANGER : Intent.NONE}
-        helperText={errors.network?.message}
+        label='URL or path'
```
Comment (elmargasimov) [inline frontend/src/components/network-import-form/network-import-form.tsx:82]: suggestion(non-blocking): shouldn't the props be double quoted to mimic more closely html? I thought this was set via eslint. https://stackoverflow.com/questions/44208081/js-quote-best-practices-es6-react-single-double-backticks

File: frontend/src/components/network-presets/network-presets.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+const hasImportedTestNetworks = (
+  testPresets: NetworkPreset[],
+  networks: string[]
+) => {
+  return testPresets.reduce<boolean>((acc, preset) => {
+    return acc || !!networks.find(n => n === preset.name)
+  }, false)
+}
+
+const itemStyles = {
```
Comment (elmargasimov) [inline frontend/src/components/network-presets/network-presets.tsx:19]: question(non-blocking): What's the reason you are not using tailwind css?

File: frontend/src/hooks/use-import-network.tsx [hunk 5]
```diff
@@ -37,6 +37,8 @@ export function useImportNetwork() {
           overwrite: force
         })
 
+        console.log(res)
```
Comment (elmargasimov) [inline frontend/src/hooks/use-import-network.tsx:40]: suggestion(non-blocking): please remove lingering console log if not required

File: frontend/src/hooks/use-import-network.tsx [hunk 6]
```diff
@@ -56,6 +58,7 @@ export function useImportNetwork() {
           throw new Error("Error: couldn't import network configuration")
         }
       } catch (err: unknown) {
+        console.log(err)
```
Comment (elmargasimov) [inline frontend/src/hooks/use-import-network.tsx:61]: suggestion(non-blocking): please remove lingering console log if not required. If required, please use console.error(err)

File: frontend/src/lib/networks.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+export type NetworkConfig = z.infer<typeof networkConfigSchema>
+
+export const networkPresetSchema = z.object({
+  name: z.string(),
+  configFileUrl: z
+    .string()
+    .url({ message: 'Invalid network configuration url.' })
+    .endsWith('.toml', {
+      message:
+        'Invalid network configuration file extension. Only ".toml" files are supported.'
```
Comment (elmargasimov) [inline frontend/src/lib/networks.ts:19]: nice 👍

File: frontend/src/routes/onboard/onboard.tsx [hunk 8]
```diff
@@ -302,16 +303,51 @@ export function OnboardWalletImport() {
 }
 
 export function OnboardNetwork() {
+  const { state, actions, dispatch } = useGlobal()
+  // The current view of the drawer
+  const [view, setView] = React.useState<'add' | 'edit' | 'manage'>('manage')
```
Comment (elmargasimov) [inline frontend/src/routes/onboard/onboard.tsx:308]: suggestion(non-blocking): 'add' | 'edit' | 'manage' is a repetition similar to above. Perhaps best to redeclare these as typescript enums

**Review by elmargasimov — DISMISSED** [review]

File: frontend/src/components/network-presets/network-presets.tsx (hunk 4 above)
Comment (notbot00) [reply frontend/src/components/network-presets/network-presets.tsx:19]: no idea, i haven't written the repo, just inherited it a few weeks ago :)

File: frontend/src/components/chrome/drawer-manage-network.tsx (hunk 1 above)
Comment (notbot00) [reply frontend/src/components/chrome/drawer-manage-network.tsx:20]: migrating the repo over to tailwind would be a massive task, and not related to the ticket.

File: frontend/src/hooks/use-import-network.tsx [hunk 9]
```diff
@@ -56,6 +57,7 @@ export function useImportNetwork() {
...
```
Comment (notbot00) [inline frontend/src/hooks/use-import-network.tsx:60]:
```suggestion
```

File: frontend/src/components/network-import-form/network-import-form.tsx (hunk 3 above)
Comment (notbot00) [reply frontend/src/components/network-import-form/network-import-form.tsx:82]: created an issue for it, as imo is not related to this PR: https://github.com/vegaprotocol/vegawallet-desktop/issues/352

File: frontend/src/components/network-import-form/network-import-form.tsx (hunk 2 above)
Comment (notbot00) [reply frontend/src/components/network-import-form/network-import-form.tsx]: changed

File: frontend/src/routes/onboard/onboard.tsx (hunk 8 above)
Comment (notbot00) [reply frontend/src/routes/onboard/onboard.tsx:308]: i'd rather skip this, as the onboarding is getting removed in the next PR related to https://github.com/vegaprotocol/vegawallet-desktop/issues/269. this is a temporary solution to leave things in a working state meanwhile

**Review by elmar-vega — APPROVED** [review]

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**